src/Runtime/OMExternalConstant.inc

```diff
@@ -0,0 +1,157 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---- OMExternalParam.cpp - OMExternalParam C/C++ Implementation ------===//
+//
+// Copyright 2023 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file contains C/C++ neutral implementation of OMExternalParam.
+//
+//===----------------------------------------------------------------------===//
+
+#include <errno.h>
+#include <inttypes.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+const int i = 1;
+#define IS_SYSTEM_LE() (!((*(const char *)&i) == 0))
+
+#define XOR(a, b) (!(a) != !(b))
+
+void checkEndianness(const char constPackIsLE) {
+  if (XOR(IS_SYSTEM_LE(), constPackIsLE)) {
+    fprintf(stderr, "Constant pack is stored in a byte order that is not "
+                    "native to this current system.");
+    exit(1);
+  }
+}
+
```

Comment:
In general, at least for ONNX, the convention is that everything written out on the disk is Little Endian. So if we are running on Big Endian, shouldn't we just do an endian conversion instead of exiting?

Reply:
ONNX is an neural exchange format and it's reasonable to use Little Endian. But for a compiled model, I am afraid that it's highly depending on the target architecture, e.g. LLVMIR contains information for a specific architecture.

Reply:
It's in general a good idea to have a FIXED endianness (in this case little endian) when the data is written out on the disk. Because this way even without any metadata to indicate the endianness of the file, you can move the data file around and use it. That was how we did it on DLC Classic.

Reply:
@tungld is this issue resolved?

Reply:
I don't see any benefit of using a default (little) endian, e.g. do you often exchange a compile model between little and big endian machines?  How often do we do this? If we only compile and run a model on a big endian machine, why we need to do byteswap for gigabytes of constants?

How about discussing about this more, and possibly another PR to solve this?

Reply:
We have no control over whether people will move their model from one arch to another or not. Without metadata, it's difficult to do things without an agreed-upon endianness for constants written on the disk. Unfortunately, the rest of the world besides z pretty much defaults to little endian on disk. I think the proper solution is that the constants should always be written out in a format that has metadata to indicate its endianness.

Reply:
Yes, totally agree with having metadata to indicate endianness. Let me do it in a follow-up PR. I'll create an issue to track some remaining works after we land this patch.